### DISCUSSION TITLE
Add to the marketplace object an entry with the required dependencies

### DISCUSSION THREAD

**Discussion by edocrippaofficial** [mia-platform/community#63] [Ideas]
Hi, I recently replaced the nginx based API gateway with Envoy and I had some problems because my project didn't have enough updated components to support the Envoy version that is in the marketplace.

In particular, the authorization-service was required at least at version 2.4.0 and the console backend at least at version 23.0.1.
This was very difficult to find, because this information is only in the changelogs of the services mentioned, and not in the Envoy one.
But it's not feasible for a developer to read all the changelogs of all services to see if they have any conflicts with a new one he wants to install.

My proposal is this: insert in the object of the marketplace an entry that indicates, to who wants to install it, if the service has any required dependencies so that, once selected within the console, it is easy to see if it is possible to install the service or if updates have to be made first.

It should be easy for those who submit a new service to the marketplace, or for those who update it, to keep the list of dependencies up to date as well.

![marketplace](https://user-images.githubusercontent.com/39826608/197201927-90652dbe-0ceb-4794-a08c-8a0f2f3baaac.png)

**Comment by edocrippaofficial** [comment]
I know that, in this example, the authorization-service is not mandatory, but the entry could be like "if you want to use xxx, it must be at least at version yyy"

**Comment by akelity** [comment]
This feature has been positively evaluated and may be considered for inclusion in the roadmap in the future, depending on priorities and overall specific needs.
